test/shared/src/main/scala/zio/test/package.scala

```diff
@@ -757,4 +758,159 @@ package object test extends CompileVariants {
           .catchAll(ZStream.succeed(_))
       }
     }
+
+  /**
+   * An implementation of `ZStream#flatMap` that supports breadth first search.
+   */
+  private[test] def flatMapStream[R, R1 <: R, A, B](
```

Comment:
Wow! I think tests are the best way to ensure correctness (it's not very easy to audit this code for correctness, at least not quickly).

Reply:
Added more tests.